### PR TITLE
zig: 0.2.0 -> 0.3.0

### PR DESCRIPTION
**Pull request by andrewrk**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by xeji** [comment]
@GrahamcOfBorg build zig

File: pkgs/development/compilers/zig/default.nix [hunk 1]
```diff
@@ -1,30 +1,23 @@
...
-    "-DZIG_LIBC_INCLUDE_DIR=${stdenv.cc.libc_dev}/include"
-    "-DZIG_LIBC_LIB_DIR=${stdenv.cc.libc}/lib"
     "-DCMAKE_BUILD_TYPE=Release"
-    "-DZIG_EACH_LIB_RPATH=On"
   ];
 
-  preConfigure = ''
-    cmakeFlags="$cmakeFlags -DZIG_LIBC_STATIC_LIB_DIR=$(dirname $(cc -print-file-name=crtbegin.o)) -DZIG_DYNAMIC_LINKER=$(cc -print-file-name=ld-linux-x86-64.so.2)"
-  '';
-
```
Comment (dtzWill) [inline pkgs/development/compilers/zig/default.nix]: These are no longer needed? :grin:

**Review by dtzWill — APPROVED** [review]

File: pkgs/development/compilers/zig/default.nix (hunk 1 above)
Comment (andrewrk) [reply pkgs/development/compilers/zig/default.nix]: Correct! This is a happy side-effect of Zig's static Linux binaries (which even work on NixOS). libc is detected at runtime.

**Comment by andrewrk** [comment]
Thanks!

**Pull request merged**